### PR TITLE
Clean up create event source docs

### LINKED ISSUE #3364: Consolidate docs for creating event sources
There are multiple docs on how to create event sources in eventing > code samples.
These docs are not very visible for users and could be consolidated better under Eventing IMHO.
- https://knative.dev/docs/eventing/samples/writing-event-source-easy-way/
- https://knative.dev/docs/eventing/samples/writing-event-source/

Tasks to do:
- [ ] Move from code samples to better named topic under eventing directly
- [ ] General tidy up of docs, add aliases etc

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3364

### CODE REVIEW

File: docs/eventing/sources/creating-event-sources/_index.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: "Creating an event source"
+weight: 100
+type: "docs"
+---
+
+You can create your own event source for use with Knative Eventing components by using the following methods:
+
+- By using a [ContainerSource](../containersource) which, which is a simple way to turn any dispatcher container into a Knative event source.
+- By using [SinkBinding](../sinkbinding), which provides a framework for injecting environment variables into any Kubernetes resource that has a `spec.template` and is [PodSpecable](https://pkg.go.dev/knative.dev/pkg/apis/duck/v1#PodSpecable).
```
Comment (abrennan89) [inline docs/eventing/sources/creating-event-sources/_index.md:10]: Either update to links under "easy way" docs, or move the easy way docs to be part of the sinkbinding and containersource docs?

File: docs/eventing/sources/creating-event-sources/_index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+- By creating your own event source controller, receiver adapter, and custom resource definition (CRD).
```
Comment (abrennan89) [inline docs/eventing/sources/creating-event-sources/_index.md:11]: Check exactly what steps are to create an event source from scratch using the sample and explain properly.
There seems to be a lot of theory mixed in with actual procedure in the doc, we should split this out properly.

File: docs/eventing/sources/creating-event-sources/writing-event-source-easy-way/_index.md [hunk 3]
```diff
@@ -1,4 +1,11 @@
-# Introduction
+---
+title: "Writing an Event Source using the easy way"
+linkTitle: "Event Source - easy way"
```
Comment (abrennan89) [inline docs/eventing/sources/creating-event-sources/writing-event-source-easy-way/_index.md:3]: Improve this, not clear

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**